### PR TITLE
fix(task-query): Fix non working filters

### PR DESCRIPTION
**Pull request by jatinriverlane**
## Changes
* FIxed the non working filters for get_all_tasks GraphQL Qhery

### CODE REVIEW

**Review by samiralavi — APPROVED** [review]

File: aqueductcore/backend/routers/graphql/resolvers/task_resolver.py [hunk 1]
```diff
@@ -30,12 +30,22 @@ async def get_tasks(
             f"Maximum allowed limit for experiments is {MAX_EXPERIMENTS_PER_REQUEST}"
         )
 
+    if filters:
+        experiment = filters.experiment
+        if str(experiment.type) != "IDType.UUID":  # type: ignore
```
Comment (str-anger) [inline aqueductcore/backend/routers/graphql/resolvers/task_resolver.py:35]: I think you still allow `experiment` to be `None`, so this is why mypy is complaining on None.type. Probably:

```suggestion
    if filters is not None:
        experiment = filters.experiment
        if experiment is not None and experiment.type != IDType.UUID:
```

File: aqueductcore/backend/routers/graphql/resolvers/task_resolver.py [hunk 2]
```diff
@@ -30,12 +30,22 @@ async def get_tasks(
...
+
     tasks = await get_all_tasks(
         user_info=context.user_info,
         db_session=context.db_session,
+        start_date=filters.start_date if filters else None,
+        end_date=filters.end_date if filters else None,
         extension_name=filters.extension_name if filters else None,
         action_name=filters.action_name if filters else None,
         username=filters.username if filters else None,
+        experiment_uuid=experiment.value if experiment else None,  # type: ignore
```
Comment (str-anger) [inline aqueductcore/backend/routers/graphql/resolvers/task_resolver.py:48]: I think `experiment` is out of scope, it is not defined it you did not pass the branch of `if filters:...`. Add `experiment = None` somewhere higher to add it to the scope.

File: aqueductcore/backend/services/task_executor.py [hunk 3]
```diff
@@ -241,8 +239,10 @@ async def get_all_tasks(  # pylint: disable=too-many-arguments
     """Get list of all tasks."""
     statement = (
         select(orm.Task)
-        .options(joinedload(orm.Task.created_by_user))
-        .options(joinedload(orm.Task.experiment))
+        .join(orm.Task.created_by_user)
+        .join(orm.Task.experiment)
+        .options(selectinload(orm.Task.created_by_user))
+        .options(selectinload(orm.Task.experiment))
```
Comment (str-anger) [inline aqueductcore/backend/services/task_executor.py:245]: I hope this does not create 2 joins =)

**Review by str-anger — CHANGES_REQUESTED** [review]
Very minor change request, and I am happy to approve. Basically cleaning mypy

File: aqueductcore/backend/services/task_executor.py (hunk 3 above)
Comment (jatinriverlane) [reply aqueductcore/backend/services/task_executor.py:245]: No, it's creating the joins once. The only change in the final query is that previously the joins were `left outer` and now joins are `inner`.

File: aqueductcore/backend/routers/graphql/resolvers/task_resolver.py (hunk 2 above)
Comment (jatinriverlane) [reply aqueductcore/backend/routers/graphql/resolvers/task_resolver.py:48]: Sure, I've added `experiment = None` to initialise `experiment` variable in `line 33`

**Review by str-anger — APPROVED** [review]

**Review by safeamiiir — APPROVED** [review]

**Pull request merged**